warden/group/manager_sql.go

```diff
@@ -123,6 +125,28 @@ func (m *SQLManager) RemoveGroupMembers(group string, subjects []string) error {
 	return nil
 }
 
+func (m *SQLManager) ListGroups(limit, offset int64) ([]string, error) {
+	if limit < 0 {
+		return nil, errors.New("limit can't be less than 0")
+	}
+
+	if offset < 0 {
+		return nil, errors.New("offset can't be less than 0")
+	}
+
+	var q []string
+
+	if limit == 0 {
+		limit = 500
+	}
+
+	if err := m.DB.Select(&q, m.DB.Rebind("SELECT id from hydra_warden_group ORDER BY id LIMIT ? OFFSET ?"), limit, offset); err != nil {
```

Comment:
I'm not sure about the order clause here, what it's goal?

Reply:
ORDER BY is required to ensure that the paging is consistent.
https://www.postgresql.org/docs/8.2/static/queries-limit.html

Reply:
Oh, I did not know that!